### ISSUE TITLE
security_nat_source - appears to be immutable for additional rules

### ISSUE THREAD

**Issue by JohnPolansky** [jeremmfr/terraform-provider-junos#281]
We are trying to use a Juniper vSRX for multiple customer VPN tunnels, the idea is we'll add new tunnels any time a customer signs up and we are using an AWS Transit gateway to share traffic between accounts behind the scenes. This means that we need to setup: 
```
set security nat static rule-set default-customer from routing-instance tgw

# Customer 1
set security nat static rule-set default-customer rule tgw-to-vpn1 match source-address 0.0.0.0/0
...

# Customer 2
set security nat static rule-set default-customer rule tgw-to-vpn2 match source-address 0.0.0.0/0
...
```
To handle NATing of the traffic for the transit gateway. The issue is, JUNOS doesn't allow you to create multiple NAT STATIC rules for the same routing instance, instead you must create "additional" rules on the existing one.

So our plan was the TF script would "add" the new rules onto the existing `default-customer` rule-set with each new VPN. However based on my reading of your `security_nat_source` it appears immutable, so u can't "add rules" you must have them all in place when you do the initial creation which obviously isn't practical in our situation.

I'm wondering if we might need a new resource to handle the rules which attaches to the `security_nat_source`.

Appreciate any insight you can provide.

**Comment by jeremmfr** [comment]
> However based on my reading of your `security_nat_source` it appears immutable, so u can't "add rules" you must have them all in place when you do the initial creation which obviously isn't practical in our situation.
> 
> I'm wondering if we might need a new resource to handle the rules which attaches to the `security_nat_source`.

I'm confused, you mean the resource `junos_security_nat_static` ?

I integrated `rule`'s setup directly in `rule-set` because `rule` have an order in `rule-set` and it is very complicated to manage if it is different resources.
But in case of `static` nat, finally the order does not seem important, as we can see on this [KB21918](https://kb.juniper.net/InfoCenter/index?page=content&id=KB21918&actp=METADATA)
```
Note: Order is not important in static NAT because all the rules are unique. (This is forced by the configuration checker.)
```
So I will take care of adding 
- a resource to add one rule on static `rule-set` nat at one time
- an option to avoid conflict with `rule` attribute in `junos_security_nat_static` resource

**Comment by JohnPolansky** [comment]
That's correct the `junos_security_nat_static` resource, sorry for the confusion. Thanks for taking a look at the enhancement.
